## Statistical Analysis of City of Seattle Staff Demographic 
By Raj Saluja (9/23/19)

## Abstract
* Analysing a dataset containing the City of Seattle Staff Demographic
* Six questions I hope to answer through this analysis 
    1. How old is the oldest person? 
    2. Hhow old is the youngest person? 
    3. What is the average age? 
    4. What is the highest hourly rate?
    5. What is the lowest hourly rate?
    6. What is the average hourly rate?

## Dataset Preparation
* Data represents the people working in Seattle 
* Contains 7 pieces of information (gender, department, age, hourly rate, regular/temporary, and employee status) about 14,194 different people 
* Created on February 25, 2019 and last updated on August 20, 2019.

In [5]:
data_file = open("City_of_Seattle_Staff_Demographics.csv", "r");

all_data = [];

for row in data_file:
    all_data.append(row);

data_file.close();

print(all_data[:5]);

['Race/Ethnicity,Sex,Department,Age,Hourly Rate,Regular/Temporary,Employee Status\n', 'Hispanic or Latino,M,Municipal Court,57,29.75,R,A\n', 'Hispanic or Latino,F,Municipal Court,39,27.18,R,A\n', 'Hispanic or Latino,M,Police Department,52,59.75,R,A\n', 'Hispanic or Latino,M,Fire Department,30,38.28,R,A\n']


Here I extracted the dataset and made it into a list. The first line of code is where  extract the data and the second line is where I make an empty list. On the third and fourth lines I add the data to the list and on the fifth line I close the list. The sixth line is where I print the first 5 items from the list of data. I did all of this to get my data ready to extract certain parts to make calculations that answer my questions. Without the data in a list, I wouldn't be able to do anything with it.

In [6]:
del all_data[0];

age = [];
wage = [];  

for item in range(len(all_data)):
    
    staff_row = all_data[item].split(",")
    
    age.append(int(staff_row[3]))
    wage.append(float(staff_row[4]))

print("List of just the age data:")
print(age)
print("\nList of just the wage data:")
print(wage)

List of just the age data:
[57, 39, 52, 30, 34, 37, 65, 29, 65, 48, 23, 41, 59, 47, 36, 43, 37, 41, 41, 55, 58, 44, 23, 22, 37, 48, 61, 18, 51, 53, 55, 63, 52, 28, 48, 34, 29, 24, 51, 38, 22, 47, 51, 54, 52, 64, 34, 19, 43, 60, 31, 68, 67, 59, 48, 52, 40, 28, 47, 24, 63, 48, 20, 32, 50, 31, 68, 52, 23, 53, 18, 52, 42, 62, 46, 41, 34, 39, 31, 31, 39, 27, 52, 21, 47, 54, 37, 28, 54, 55, 47, 31, 60, 44, 40, 32, 37, 52, 62, 33, 46, 66, 56, 24, 28, 53, 50, 17, 61, 23, 35, 32, 45, 17, 17, 43, 54, 60, 31, 34, 17, 33, 62, 40, 48, 27, 63, 35, 48, 44, 60, 25, 35, 19, 54, 34, 38, 70, 53, 42, 43, 50, 36, 47, 27, 49, 35, 34, 64, 28, 59, 19, 34, 28, 30, 38, 44, 35, 47, 49, 53, 30, 59, 39, 38, 42, 32, 26, 59, 53, 41, 52, 26, 55, 34, 54, 34, 53, 29, 61, 58, 46, 44, 34, 48, 63, 48, 59, 16, 19, 65, 29, 72, 33, 32, 31, 48, 29, 17, 37, 58, 47, 63, 29, 40, 49, 57, 37, 32, 36, 51, 36, 53, 35, 47, 25, 61, 50, 37, 43, 53, 65, 26, 48, 62, 31, 34, 41, 67, 72, 59, 59, 34, 44, 22, 33, 39, 45, 62, 30, 43, 62, 45, 

Here I extracted the age and hourly rate data from the dataset and made them into two seperate lists. I'm not going to be using the other data to answer my questions, so I didn't need to extract them. This makes my data completely ready to use for my calculations because I can make specific calculations about each part of the data. On the first line I removed the header of the column from my data because I won't need that when making calculations. On the second and third lines I made two seperate lists: one for the age and one for the hourly rate. On the fourth line I looped through every row in the dataset and on the fifth I split the string of data by comma so each pieace of data became its own string. On the sixth and seventh lines I made my lists for age and hourly rate data just the age and hourly rate data instead of the entire dataset. On the eight, ninth, tenth, and elevnth lines I printed the age and hourly rate data.

## Data Modeling

In [13]:
count = len(age)
total = sum(age)
avg=total/count
print("Average age is", avg, "years old")

Average age is 44.53395801042694 years old


Here I calculated the average age. On the first line I found the total number of pieces of data (the total number of ages) and on the second line I found the sum of all those ages. In order to find the average, you simply need to divide the sum by the total so I did that on the third line and printed it on the fourth. This answered my question about what the average age is.

In [14]:
minAge = min(age)
maxAge = max(age)

print("The youngest person is", minAge, "and the oldest person is", maxAge)

The youngest person is 14 and the oldest person is 92


Here I found the age of the oldest person and the age of the youngest person. In order to find the youngest person you need to find the minimum age, which I do on the first line when I make a varible for that age. In order to find the oldest person you need to find the maximum age, which I do on the second line when i make a varible for that age. On the third line I print those ages to get the result. This answered my questions about how old the oldest and youngest people are.

In [12]:
count2 = len(wage)
total2 = sum(wage)
avg2=total2/count2
print("Average hourly rate is", avg2, "dollars per hour")

Average hourly rate is 40.03381942370063 dollars per hour


Here I calculated the average hourly rate. On the first line I found the total number of pieces of data (the total number of hourly rates) and on the second line I found the sum of all those hourly rates. In order to find the average, you simply need to divide the sum by the total so I did that on the third line and printed it on the fourth. This answered my question about what the average hourly rate is.

In [14]:
minWage = min(wage)
maxWage = max(wage)

print("The lowest hourly rate is", minWage, "dollars per hour and the largest hourly rate is", maxWage, "dollars per hour")

The lowest hourly rate is 5.53 dollars per hour and the largest hourly rate is 162.8353 dollars per hour


Here I found the highest hourly rate and lowest hourly rate. In order to find the lowest hourly rate you need to find the minimum age, which I do on the first line when I make a varible for that rate. In order to find the highest hourly rate you need to find the maximum hourly rate, which I do on the second line when i make a varible for that rate. On the third line I print those rates to get the result. This answered my questions about what the highest and lowest hourly rates are.

## Data Analysis and Conclusions

The average age was arouund 44 years old, and this was quite surprising. I expected this number to be a lot smaller because I know there are many people in their 20s that work, whereas I thought there wouldn't be many people older than 60 working. This average makes it seem like there are more people over the age of 50 and possibly 60 working than there are people in their 20s and early 30s. I think this might be because a lot of people are deciding to go to graduate school so they can get better jobs, which would prevent younger people from getting jobs. This is an area for future exploration.

The youngest person was 14 years old, and that was very surprising. I know that 14 is the youngest a person can be to work in the state of Washington, so this is legal, but it's incredibly surprising that this person is working for the city. Most 14 year olds work at stores or movie theaters, not the city. I wondered if this person is a temp or a regular worker, I wondered what job they had and I wondered what their hourly rate is because they are so young. I looked into this and I found that they are working in the library, so they are likely just the person who checks out books, which is a relativley normal job. They are a regular employee who makes 16 dollars an hour. The hourly rate makes sense because they are too young to make much more than the minimum wage of 15 dollars per hour. The maximum age was 92 years old, which was even more shocking. I can't even imagine somebody 92 years old working. This brought up some questions. First, what job are they doing? There probably aren't many jobs that a 92 year old could do so I would like to know. Second, are they a full-time worker? That workload seems like it would be too large for a 92 year old so I would assume they are a temporary employee. I looked into this and found that this person is a temp who makes 16.11 dollars per hour and works in Parks and Recreation. I'm surprised at how low the hourly rate is because I would have assumed that somebody so old would only work if they got paid a lot. The fact that they're a temp is not surprising. They probably work in a section of Parks and Recreation that doesn't require a ton of physical activity. 

The average hourly rate was around 40 dollars per hour. I was surprised by this because I expected the hourly rate to be smaller because I didn't think the jobs for the city were incredibly high paying. I also assumed that most of these jobs would be jobs that made close to minimum wage. It is likely that there are a few very high paying jobs that are bringing the average up. This is an area for future exploration.

The lowest hourly rate was 5.53 dollars which was incredibly surprising because it's below minimum wage (15 dollars per hour). I thought the only explanation for this is that it was likely a typo. I wanted to make sure of this so I went to the dataset to check, but I found that there are sic people making exactly 5.53 dollars per hour, and they all work in the same department: Civil Service Commissions. The next lowest paid person was making 16 dollars per hour, which is above minimu wage. The only explanations for these people all making 5.53 dollars per hour is either that there was a typo for the entire department or that department is special and doesn't require employees to be paid minimum wage. It's most likely that it was a typo, but this is an area for future exploration. The highest hourly rate was 162.8353 dollars per hour, which I would have found incredibly surprising but I expected it because of the average. I knew that in order for the average to be so high there must have been a few people with very high hourly rates and I was right because the hightest rate is absurdly high. I didn't think anybody made this much money per hour, especially not people working for the city, so this made me wonder what job this person has. I looked into it and found that they work for Seattle City Light. I think the most logical explanation is that they are an engineer whose skill set is in very high demand which is why they make so much money. This is an area for future exploration.

Overall, the results of my calculations were incredibly surprising, but I was able to come up with some reasons as to why they are what they are after looking further. I learned a lot from this data and there are a lot of things to explore in the future.

## Acknowledgements

I would like to acknowledge Ms. Sconyers and Mark for helping me throughout this lab, especially with my code. They were a big help.